examples/examples/postgres-partial.rs

```diff
@@ -0,0 +1,102 @@
+use std::sync::Arc;
+use tokio::task;
+
+use datafusion::{
+    catalog::schema::SchemaProvider,
+    error::Result,
+    execution::context::{SessionContext, SessionState},
+};
+use datafusion_federation::{FederatedQueryPlanner, FederationAnalyzerRule};
+use datafusion_federation_sql::{MultiSchemaProvider, SQLFederationProvider, SQLSchemaProvider};
+use datafusion_federation_sql::connectorx::CXExecutor;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let state = SessionContext::new().state();
+    // Register FederationAnalyzer
+    // TODO: Interaction with other analyzers & optimizers.
+    let state = state
+        .add_analyzer_rule(Arc::new(FederationAnalyzerRule::new()))
+        .with_query_planner(Arc::new(FederatedQueryPlanner::new()));
+
+let df = task::spawn_blocking(move || {
+    // Register schema
+    let pg_provider = async_std::task::block_on(create_postgres_provider(vec!["class"], "conn1")).unwrap();
+    let pg_col_provider = async_std::task::block_on(create_postgres_provider(vec!["teacher"], "conn2")).unwrap();
+    let provider = MultiSchemaProvider::new(vec![
+        pg_provider,
+        pg_col_provider,
+    ]);
+
+    overwrite_default_schema(&state, Arc::new(provider)).unwrap();
+
+    // Run query
+    let ctx = SessionContext::new_with_state(state);
+    let query = r#"SELECT class.name AS classname, teacher.name as teachername from class join teacher on class.id = teacher.class_id;"#;
+        let df = async_std::task::block_on(ctx.sql(query)).unwrap();
+
+        df
+    }).await.unwrap();
+
+    task::spawn_blocking(move || { async_std::task::block_on(df.show()) }).await.unwrap()
+}
+
+async fn create_postgres_provider(
+    known_tables: Vec<&str>,
+    context: &str,
+) -> Result<Arc<SQLSchemaProvider>> {
+    let dsn = "postgresql://suriya-retake:password@localhost:28816/pg_analytics".to_string();
```

Comment:
Naturally this won't work for anyone. I wonder what the best way is to handle this. Including the example makes sense to me. I'm not sure if it's worth trying to make it runnable. Maybe we can just make the url a bit more generic.

Reply:
Perhaps for integration testing we could create some Dockerfiles + docker compose set up that will deploy databases such as postgres which we can federate against. Some examples can be flagged as requiring running `docker compose up` before they will function.